src/ZoneServer/World/Actors/CombatEntities/Components/CooldownComponent.cs

```diff
@@ -49,7 +49,7 @@ public void Start(CooldownId cooldownId, TimeSpan duration)
 				}
 			}
 
-			if (this.Entity is Character character)
+			if (this.Entity is Character character && character is not DummyCharacter)
```

Comment:
You're mixing conventions of checking for dummy characters before calling senders and inside them, as well as mixing type and null checks. I do understand what led to this, but it would be better to stay consistent. Better yet would be to not need this distinction.